### DISCUSSION TITLE
Map performance

### DISCUSSION THREAD

**Discussion by Ammar441** [LuckyPennySoftware/AutoMapper#4018] [NA]
I have an ASP.NET Core MVC project using Razor pages. I'm using a repository and after that using Automapper to convert to a view model. The query in the repository is fast, but when I use Automapper to convert the data returned on a listViewModel, it is very slow when I get back more than 1000 rows.

This is a function to get all rows in the repository:

### Source/destination types

```csharp
public IQueryable<ModelName> GetAllAsync()
{
    return context.ModelName
                  .OrderByDescending(m => m.CreateOn)
                  .Include(m => m.Customer)
                  .Include(m => m.Town);
}
```

### Mapping configuration

```csharp
public EvAutoMapper()
        {
            CreateMap<Model, ModelDto>().ReverseMap();
}
```

### Steps to reproduce

```csharp
var viewModelList = list.Select(m => _mapper.Map<ViewModel>(m))
                        .ToList();

//also I'm try this 
var viewModelList =  _mapper.Map<IQueryable<ModelName>,IEnumerable<ModelDto>>(source)
                        .ToList();
```
 

I'm tired to improve performance

**Comment by lbargaoanu** [comment]
https://docs.automapper.org/en/stable/Queryable-Extensions.html
